### PR TITLE
feat(pkg/*) Add local cluster domain to mesh service.

### PR DESCRIPTION
**Pull request by steeling**
This also requires adding it to the envoy cluster

Co-authored-by: brunoterkaly <bterkaly@microsoft.com>

Signed-off-by: Sean Teeling <seanteeling@microsoft.com>

#3444

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| Documentation              | [ ] |
| Install                    | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Ingress                    | [ ] |
| Egress                     | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| SMI Policy                 | [ ] |
| Sidecar Injection          | [ ] |
| Security                   | [ ] |
| Upgrade                    | [ ] |
| Tests                      | [ ] |
| CI System                  | [ ] |
| Demo                       | [ ] |
| Performance                | [ ] |
| Other                      | [x ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?
No

1. Is this a breaking change?
No

### CODE REVIEW

**Review by snehachhabria — APPROVED** [review]

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -208,3 +208,11 @@ const (
 	// byte in a TCP connection.
 	ProtocolTCPServerFirst = "tcp-server-first"
 )
+
+// ClusterDomain is a type for a multicluster
+type ClusterDomain string
```
Comment (shashankram) [inline pkg/constants/constants.go:213]: Implement a `String()` method receiver for this type so we can use `LocalDomain.String()` vs `string(LocalDomain)` for consistency.

**Review by shashankram — APPROVED** [review]

File: pkg/constants/constants.go (hunk 1 above)
Comment (steeling) [reply pkg/constants/constants.go:213]: done!

File: pkg/envoy/secrets/secrets.go [hunk 2]
```diff
@@ -30,7 +31,7 @@ func UnmarshalSDSCert(str string) (*SDSCert, error) {
 		return nil, errInvalidCertFormat
 	}
 
-	// Make sure the slices are not empty. Split might actually leave empty slices.
+	// Make sure the slices are not empty. Split might actually leave empervices.y slices.
```
Comment (snehachhabria) [inline pkg/envoy/secrets/secrets.go:34]:
```suggestion
	// Make sure the slices are not empty. Split might actually leave empty slices.
```

Comment (steeling) [reply pkg/envoy/secrets/secrets.go]: done!

**Pull request merged**